extensions/platform-http/runtime/src/main/java/org/apache/camel/quarkus/component/platform/http/runtime/QuarkusPlatformHttpConsumer.java

```diff
@@ -0,0 +1,376 @@
+                // use an iterator as there can be multiple values. (must not use a delimiter)
+                final Iterator<?> it = ObjectHelper.createIterator(value, null);
+                String firstValue = null;
+                List<String> values = null;
+                while (it.hasNext()) {
+                    final String headerValue = tc.convertTo(String.class, it.next());
+                    if (headerValue != null
+                            && !headerFilterStrategy.applyFilterToCamelHeaders(key, headerValue, exchange)) {
+                        if (firstValue == null) {
+                            firstValue = headerValue;
+                        } else {
+                            if (values == null) {
+                                values = new ArrayList<String>();
+                                values.add(firstValue);
+                            }
+                            values.add(headerValue);
+                        }
+                    }
+                }
+                if (values != null) {
+                    response.putHeader(key, values);
+                } else if (firstValue != null) {
+                    response.putHeader(key, firstValue);
+                }
+            }
+        }
+
+        Object body = message.getBody();
+        final Exception exception = exchange.getException();
+
+        if (exception != null) {
+            // we failed due an exception so print it as plain text
+            final StringWriter sw = new StringWriter();
+            final PrintWriter pw = new PrintWriter(sw);
+            exception.printStackTrace(pw);
+
+            // the body should then be the stacktrace
+            body = ByteBuffer.wrap(sw.toString().getBytes(StandardCharsets.UTF_8));
+            // force content type to be text/plain as that is what the stacktrace is
+            message.setHeader(Exchange.CONTENT_TYPE, "text/plain; charset=utf-8");
+
+            // and mark the exception as failure handled, as we handled it by returning it as the response
+            ExchangeHelper.setFailureHandled(exchange);
+        }
+
+        // set the content type in the response.
+        final String contentType = MessageHelper.getContentType(message);
+        if (contentType != null) {
+            // set content-type
+            response.putHeader("Content-Type", contentType);
+        }
+        return body;
+    }
+
+    void writeResponse(RoutingContext ctx, Exchange camelExchange, HeaderFilterStrategy headerFilterStrategy) {
+        final Object body = toHttpResponse(ctx.response(), camelExchange.getMessage(), headerFilterStrategy);
+
+        final HttpServerResponse response = ctx.response();
+        if (body == null) {
+            QuarkusPlatformHttpEngine.LOG.tracef("No payload to send as reply for exchange: %s", camelExchange);
+            response.putHeader("Content-Type", "text/plain; charset=utf-8");
+            response.end("No response available");
```

Comment:
Isnt there a HTTP STATUS 204 (or something like that for -- OK but no content)

Reply:
I think I copied this from the Undertow component. Let me check what the other HTTP components do.

Reply:
So yes, this behavior is from Undertow. `org.apache.camel.http.common.DefaultHttpBinding.doWriteResponse(Message, HttpServletResponse, Exchange)` does not write anything if the body is null. 

The status code at this location can be both success or failure, so it would not always be correct to change it to 204.

Reply:
I personally find the current code good enough, but adding a note to my TODO list that this case should be covered by a test. BTW, which of the existing HTTP components is the most up to date? - i.e. where to look for the desired behavior?

Reply:
Yeah ideally we should get the http components aligned. They should IMHO not return a hardcoded text, eg this is in english, then chinese users get english text etc. But lets log a ticket upstream at Apache Camel

Reply:
Filed https://github.com/apache/camel-quarkus/issues/220